Add description of the Alex Boev's HN PWA implementation

alexboev opened the pull request:
Hi! I'd like to suggest my own implementation of the Hacker News PWA.

kristoferbaxter commented:
A few items I noticed on quick glance.

1.  'React HN PWA (Alex Boev)' is a new pattern for Titles, would you mind if we stuck to a bit of the convention and went with something like 'React HN'?

2. We've begun to settle on a common pagination system for implementations. You can see this pattern on quite a few entries, including: 
https://nextjs-hn.chrisdwheatley.com/
https://angularhn.sebastian-mueller.net/

3. I'm seeing a development build of React in the production deployed version.
![screen shot 2017-07-24 at 9 20 57 am](https://user-images.githubusercontent.com/61764/28533303-6cf2b9aa-7051-11e7-9da9-e64289198cd1.png)

alexboev commented:
Kristofer, thanks a lot for your comments! I'll fix everything soon. As for the name, I've chosen such a strange pattern in order to avoid another simple "ReactHN". I'll change it of course, but it appears that both the cases with the name and the pagination deserve to be added to the specs to avoid further confusion. What do you think?

kristoferbaxter commented:
Checked WPT/Easy for timing information as well.

EM_3G, **4.7s**
https://www.webpagetest.org/result/170724_KK_288d3eb495d5a2c788f1b24d3b69dd71/

Faster 3G, **3.9s**
https://www.webpagetest.org/result/170724_5H_0a8773a40469a1da1dd574825f5e0787/

kristoferbaxter commented:
@addyosmani Alex raises a good point, I agree we should update the specs.

alexboev commented:
BTW, kristofer, can you please point me to these numbers - 4.7 and 3.9? I couldn't find exactly these. :) I'd like to know which parameters to watch during tests. And there is a little implementation detail - there is a primitive in-memory cache on my "server" which allows to avoid doing 30 request per page. So I think that fugures should be better if you are running the test right after manual refreshing of the page which you are trying to test.

kristoferbaxter commented:
Sure @alexboev!

On the first report (https://www.webpagetest.org/result/170724_KK_288d3eb495d5a2c788f1b24d3b69dd71/) there is an icon for the Lighthouse report for that specific run.

![screen shot 2017-07-24 at 10 25 14 am](https://user-images.githubusercontent.com/61764/28535781-6aeeeabc-705a-11e7-9674-a136bdddb841.png)

Clicking there will show you the lighthouse report generated when WPT executed this specific test. (https://www.webpagetest.org/lighthouse.php?test=170724_KK_288d3eb495d5a2c788f1b24d3b69dd71&run=3)

The value you're looking for is under performance labeled "Consistently Interactive"
![screen shot 2017-07-24 at 10 26 35 am](https://user-images.githubusercontent.com/61764/28535840-a82de8ec-705a-11e7-90b4-c92f00519dd2.png)

There is a ton of great information available via WPT and Lighthouse – go explore!

alexboev commented:
@kristoferbaxter, I've done all the changes and updated stats. I'm not sure why the new stats are so considerably worse. The changes are not so dramatic.

kristoferbaxter commented:
I am seeing an unminified build of React used when I access the application. Are you working on changes?

![screen shot 2017-07-26 at 9 06 31 am](https://user-images.githubusercontent.com/61764/28631355-c17b0266-71e1-11e7-9337-6ac2c07fd1f2.png)

![screen shot 2017-07-26 at 9 07 33 am](https://user-images.githubusercontent.com/61764/28631391-e85a7ac4-71e1-11e7-82c7-5f1193492c14.png)

alexboev commented:
Oh, that was my fault - I have forgotten to minify. I've updated the bundle now.

alexboev commented:
I've noticed that after minification the performance figures are better - https://www.webpagetest.org/lighthouse.php?test=170726_GH_53d626acc9af7552dcb44928f91278ef&run=1 I'm not sure is it related or not

alexboev commented:
Hi, @kristoferbaxter and other guys! It would be nice to arrive at some finish line. Are there any problems now with regard to my implementation?

housseindjirdeh commented:
Apologies @alexboev for not hopping on to take a look at this sooner! 

Thanks a million for this, a couple of things I want to point out however:

1. You mentioned that after minifying your main bundle your performance numbers improved. Running a few WPT entries myself and your numbers are looking good:

![image](https://user-images.githubusercontent.com/12476932/29011740-774c96e8-7b03-11e7-87b6-6016949c88db.png)

Any chance you can update the numbers you currently have in your submission? You may have forgotten to.

2. So it looks like you're leveraging an App Shell cached to your service worker but it still looks like your REST calls are failing without a network connection. Doesn't seem to be that you're relying on any dynamic caching.

![image](https://user-images.githubusercontent.com/12476932/29011786-d2a9ef86-7b03-11e7-9cf8-bad7b04144b6.png)

Have you thought of including this functionality in? Will definitely make things more concrete allowing your app to function offline with pages previously visited.

3. For each implementation in HNPWA, they're usually differentiated by a specific name and/or logo. Have you thought of having the name in your application in the top left reflect your current title (`React-Redux HN` or `Alex Boev HN`) instead of `Hacker News`?

4. Although we haven't laid out specific styling requirements for the applications, it would probably be a little nicer if you can add a little padding to corners and between items in your application to give things more breathing room (especially for mobile).

![image](https://user-images.githubusercontent.com/12476932/29011846-7b129a6a-7b04-11e7-97f6-8c2706b2404b.png)

Would love to hear your thoughts on the above and will be happy to assist in getting those items covered. Once we do, will be glad to have this in HNPWA :) :)

alexboev commented:
Hi @housseindjirdeh!

Thank you very much for your feedback!

I've tried to implement all the fixes you suggested and here's the result:

- minor styling changes concerning paddings and margins
- offline mode detection with notification on page
- offline cache for data
- new logo and title
- updated screenshot inside mobile phone mockup
- updated stats on page speed tests

I hope I was able to address all the concerns you've mentioned.

housseindjirdeh commented:
thanks a million @alexboev, will do a run through once more later today/tomorrow 🙌

alexboev commented:
Hi @housseindjirdeh!

Is there any chance to get your feedback again? I'd like to move this little "project" to its end. :) It appears to me that everything is OK now.
